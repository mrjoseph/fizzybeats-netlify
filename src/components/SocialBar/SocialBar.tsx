import React from 'react';
import Typography from '@material-ui/core/Typography';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';
import instagram from './images/iconmonstr-instagram-11.svg';
import soundcloud from './images/iconmonstr-soundcloud-1.svg';
import facebook from './images/iconmonstr-facebook-4.svg';
import youtube from './images/iconmonstr-youtube-6.svg';
const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
    },
    paper: {
      padding: theme.spacing(2),
      textAlign: 'center',
      background: 'none',
      border: 0,
      borderRadius: 0,
      boxShadow: 'none',

    },
  }),
);


const SocialBar = () : JSX.Element | null => {
    const classes = useStyles();
    const socialMedaLinks = [
        {
            title: 'instagram',
            image: instagram,
            url: 'https://www.instagram.com/fizzy_beats/',
        },
        {
            title: 'Soundcloud',
            image: soundcloud,
            url: 'https://soundcloud.com/fizzybeats',
        },
        {
            title: 'Facebook',
            image: facebook,
            url: 'https://www.facebook.com/FizzybeatsDnB/',
        }
        ,
        {
            title: 'Youtube',
            image: youtube,
            url: 'https://www.youtube.com/user/fizzybeatsDnB',
        }
    ];
    return (
    <React.Fragment>
        <div className={classes.root}>
            <Typography component="h4">
            LISTEN TO FIZZY BEATS ON:
            </Typography>
            <Grid container spacing={3}>
                {socialMedaLinks.map(({ title, image, url}) => {
                    return (
                        <Grid item xs={6} sm={3} key={title}>
                            <Paper className={classes.paper}>
                                <a href={url} target="_blank">
                                <img src={image} alt={title} />
                                </a>
                            </Paper>
                    </Grid>
                    )

                })}
            </Grid>
        </div>
    </React.Fragment>
    )
}

export default SocialBar;