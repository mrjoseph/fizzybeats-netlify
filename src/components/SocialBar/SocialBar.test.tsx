  
import React from 'react';
import SocialBar from './SocialBar';
import renderer from 'react-test-renderer';
describe('SocialBar', () => {
    describe('snapshot', () => {
        it('renders correctly', () => {
            interface IsocialBar {}    
            const props:IsocialBar = {}
            const tree = renderer.create(<SocialBar {...props}/>).toJSON();
            expect(tree).toMatchSnapshot();
          });
    })
});