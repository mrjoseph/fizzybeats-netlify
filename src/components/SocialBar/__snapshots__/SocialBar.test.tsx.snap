// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialBar snapshot renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <h4
    className="MuiTypography-root MuiTypography-body1"
  >
    LISTEN TO FIZZY BEATS ON:
  </h4>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 makeStyles-paper-2 MuiPaper-rounded"
      >
        <a
          href="https://www.instagram.com/fizzy_beats/"
          target="_blank"
        >
          <img
            alt="instagram"
            src="iconmonstr-instagram-11.svg"
          />
        </a>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 makeStyles-paper-2 MuiPaper-rounded"
      >
        <a
          href="https://soundcloud.com/fizzybeats"
          target="_blank"
        >
          <img
            alt="Soundcloud"
            src="iconmonstr-soundcloud-1.svg"
          />
        </a>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 makeStyles-paper-2 MuiPaper-rounded"
      >
        <a
          href="https://www.facebook.com/FizzybeatsDnB/"
          target="_blank"
        >
          <img
            alt="Facebook"
            src="iconmonstr-facebook-4.svg"
          />
        </a>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 makeStyles-paper-2 MuiPaper-rounded"
      >
        <a
          href="https://www.youtube.com/user/fizzybeatsDnB"
          target="_blank"
        >
          <img
            alt="Youtube"
            src="iconmonstr-youtube-6.svg"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
